test/unittests.jl

```diff
@@ -46,6 +46,41 @@ end
     @test Base.Array(x_device)≈x_expected rtol=1e-1
 end
 
+@testset "LaunchSpec" begin
+    N = 100
+    dims = (N)
+    a = round.(rand(Float32, dims) * 100)
+    a_expected = a .+ 5.0
+    a_device = JACC.array(a)
+    JACC.parallel_for(JACC.launch_spec(; threads = 1000), N,
+        (i, a) -> begin
+            @inbounds a[i] += 5.0
+        end, a_device)
+    @test Base.Array(a_device)≈a_expected rtol=1e-5
+
+    A = JACC.ones(Float32, N, N)
+    B = JACC.ones(Float32, N, N)
+    C = JACC.zeros(Float32, N, N)
+    JACC.parallel_for(JACC.launch_spec(; threads = (16, 16)),
+        (N, N), (i, j, A, B, C) -> begin
+            @inbounds C[i, j] = A[i, j] + B[i, j]
+        end,
+        A, B, C)
+    C_expected = Float32(2.0) .* ones(Float32, N, N)
+    @test Base.Array(C)≈C_expected rtol=1e-5
+
+    A = JACC.ones(Float32, N, N, N)
+    B = JACC.ones(Float32, N, N, N)
+    C = JACC.zeros(Float32, N, N, N)
+    JACC.parallel_for(JACC.launch_spec(; threads = (4, 4, 4)),
```

Comment:
I guess this is required to be a first argument since we are using Varargs. Another overloaded version could be similar to CUDA.jl (since AMDGPU.jl now follows it):
` JACC.@lauch_spec  threads = (16, 16) blocks = (1,1) sync=false  JACC.parallel_for(N, f, x...)`

Reply:
I'll open an issue for this feature.